applications/FSIapplication/python_scripts/convergence_accelerator_factory.py

```diff
@@ -2,52 +2,73 @@
 
 import KratosMultiphysics
 import KratosMultiphysics.FSIApplication as KratosFSI
+try:
+    import KratosMultiphysics.TrilinosApplication as KratosTrilinos
+    have_trilinos = True
+except ImportError:
+    have_trilinos = False
+
+_aitken_defaults = KratosMultiphysics.Parameters("""{   "solver_type"       : "Relaxation",
+                                                        "acceleration_type" : "Aitken",
+                                                        "w_0"               : 0.825  }""")
+
+_mvqn_defaults = KratosMultiphysics.Parameters("""{ "solver_type" : "MVQN",
+                                                    "w_0"         : 0.825   }""")
+
+_mvqn_defaults_recursive = KratosMultiphysics.Parameters("""{ "solver_type" : "MVQN_recursive",
+                                                              "w_0"         : 0.825,
+                                                              "buffer_size" : 7           }""")
 
 def CreateConvergenceAccelerator(configuration):
-    
+
     if(type(configuration) != KratosMultiphysics.Parameters):
         raise Exception("Input is expected to be provided as a Kratos Parameters object.")
-    
+
     convergence_accelerator_type = configuration["solver_type"].GetString()
-    
+
     if(convergence_accelerator_type == "Relaxation"):
-        
-        defaults_relaxation = KratosMultiphysics.Parameters("""{
-                                                                "solver_type"       : "Relaxation",
-                                                                "acceleration_type" : "Aitken",
-                                                                "w_0"               : 0.825
-                                                               }""")
-                                                               
-        configuration.ValidateAndAssignDefaults(defaults_relaxation)        
-        
+
+        configuration.ValidateAndAssignDefaults(_aitken_defaults)
+
         if (configuration["acceleration_type"].GetString() == "Aitken"):
-            convergence_accelerator = KratosFSI.AitkenConvergenceAccelerator(configuration["w_0"].GetDouble())       
```

Comment:
why don't you pass the parameters in the constructor? it is designed for this, and it will allow you simplifying a lot this factory.

Reply:
note that the defaults and the validations should be done within the constructor of each separate accelerator

Reply:
You are right, it can be done just creating a new constructor with a unique Parameters object as argument. The point is how we can firstly perform the ValidateAndAssignDefaults() in case there is a const member variable...

Reply:
I would say in extreme.case you can use a const_cast.

In any case the advantage of having everything in the constructor is large, for many reasons.

Reply:
Done!